src/client/datascience/kernel-launcher/kernelProcess.ts

```diff
@@ -349,7 +350,6 @@ export class KernelProcess implements IKernelProcess {
             // a different process tries to open it.
             const tempFile = await this.fileSystem.createTemporaryLocalFile('.json');
             this.connectionFile = tempFile.filePath;
-            await tempFile.dispose();
```

Comment:
This is wrong, we need the file to exist, the next line we attempt to write to the file.
What is happening is sometimes the file is created and sometimes its deleted after we connect.

@rchiodo (found this based on the error messages logged by the kernel after adding `--debug`)
**New changes after approval**

Reply:
Should the dispose happen later then? Like maybe in the shutdown of the kernel?

Reply:
Will do that as part of https://github.com/microsoft/vscode-jupyter/issues/8827